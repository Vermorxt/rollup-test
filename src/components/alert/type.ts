import { Ui_colors } from '../_types/colors/ui_colors'
import { HTMLProps, MouseEventHandler } from 'react'

export interface Ui_AlertProps extends HTMLProps<HTMLDivElement>, Ui_colors {
  noIcon?: boolean
  onDenyButtonText?: string
  onAcceptButtonText?: string
  onDeny?: MouseEventHandler<HTMLButtonElement> | undefined
  onAccept?: MouseEventHandler<HTMLButtonElement> | undefined
}
