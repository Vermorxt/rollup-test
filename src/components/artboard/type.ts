import { Ui_colors } from './../_types/colors/ui_colors'
import { HTMLProps } from 'react'

export interface Ui_ArtboardProps extends HTMLProps<HTMLSpanElement>, Ui_colors {
  phone1?: boolean
  phone2?: boolean
  phone3?: boolean
  phone4?: boolean
  phone5?: boolean
  phone6?: boolean
  horizontal?: boolean
  demo?: boolean
}
