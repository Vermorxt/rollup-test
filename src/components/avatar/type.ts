import { HTMLProps } from 'react'
import { Ui_Sizes } from '../_types/sizes/sizes'

interface Ui_Avatar_radius {
  roundedXl?: boolean
  rounded2xl?: boolean
  rounded3xl?: boolean
  roundedLg?: boolean
  roundedFull?: boolean
}

interface Ui_Avatar_ring {
  ring?: boolean
  ringPrimary?: boolean
  ringSecondary?: boolean
  ringOffset?: '0' | '1' | '2' | '4' | '8'
  ringOffsetBackground?: string // NOTE: https://tailwindcss.com/docs/ring-offset-color
}

interface Ui_Avatar_shapes {
  squircle?: boolean
  hexagon?: boolean
  triangle?: boolean
}

interface Ui_Avatar_online {
  online?: boolean
  offline?: boolean
}

export interface Ui_AvatarProps
  extends HTMLProps<HTMLDivElement>,
    Ui_Sizes,
    Ui_Avatar_radius,
    Ui_Avatar_shapes,
    Ui_Avatar_online,
    Ui_Avatar_ring {
  src?: string
  usePlaceHolder?: string // NOTE: content of last avatar item (e.g.: +99)
}
