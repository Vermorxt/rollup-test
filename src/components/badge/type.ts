import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'

export type TooltipPositionProps = 'top' | 'left' | 'bottom' | 'right'

export interface Ui_Badge_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Badge_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_BadgeProps
  extends HTMLProps<HTMLSpanElement>,
    Ui_colors,
    Ui_Sizes,
    Ui_Badge_variants,
    Ui_Badge_singleClass {
  wide?: boolean
  full?: boolean
}
