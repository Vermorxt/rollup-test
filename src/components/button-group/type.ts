import { HTMLProps } from 'react'

export interface Ui_ButtonGroupProps extends HTMLProps<HTMLDivElement> {}
