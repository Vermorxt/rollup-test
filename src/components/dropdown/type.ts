import { HTMLProps, ReactElement, ReactNode } from 'react'
import { AnyType } from '../../_types/AnyType'
import { Ui_Button_singleClass, Ui_Button_sizes, Ui_Button_variants } from '../button/type'
import { Ui_colors } from '../_types/colors/ui_colors'

export interface Ui_DropdownPosition {
  top?: boolean
  left?: boolean
  end?: boolean
  right?: boolean
}
export interface Ui_DropdownType {
  full?: boolean
  card?: boolean
}
export interface Ui_DropdownAction {
  hover?: boolean
  open?: boolean
}
export interface Ui_DropdownProps
  extends Omit<HTMLProps<Element>, 'content'>,
    Ui_colors,
    Ui_DropdownPosition,
    Ui_DropdownType,
    Ui_Button_sizes,
    Ui_Button_variants,
    Ui_Button_singleClass,
    Ui_DropdownAction {
  children?: ReactNode
  className?: string
  tabIndex?: number
  content?: ReactElement<AnyType>
}
