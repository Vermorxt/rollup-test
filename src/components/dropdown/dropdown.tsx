import { getClassNamesFromAttributes } from '../_utils/css-class-generator'
import { FC, forwardRef } from 'react'
import { Ui_DropdownProps } from './type'
import React from 'react'

export const Ui_Dropdown: any = forwardRef<
  HTMLDivElement,
  Ui_DropdownProps & { Label: FC<Ui_DropdownProps>; Content: FC<Ui_DropdownProps> }
>((props, ref) => {
  const {
    children,
    className,
    tabIndex,
    label,
    content,
    style,
    loading,
    wide,
    block,
    full,
    square,
    large,
    medium,
    small,
    tiny,
    circle,
    disabled,
    noAnimation,
    allowFullScreen,
    info,
    warning,
    success,
    error,
    primary,
    secondary,
    accent,
    ghost,
    link,
    neutral,
    outline,
    ...rest
  } = props

  const convertAttributeToSingleClass = [['full'], ['w-full']]

  const singleClasses = getClassNamesFromAttributes({
    names: rest,
    convert: convertAttributeToSingleClass,
    withoutPrefix: [],
    addPrefix: null,
  })

  const convertAttributeToClassName = [['rounded'], ['rounded-t-box rounded-b-box']]

  const classAttributes = getClassNamesFromAttributes({
    names: rest,
    convert: convertAttributeToClassName,
    withoutPrefix: [],
    addPrefix: 'dropdown',
  })
  const defaultClassNames = `dropdown ${`${classAttributes} ${singleClasses} ${(className as string) || ''}`}`

  if (!label && !content) {
    return (
      <div ref={ref} style={style} className={defaultClassNames}>
        {children}
      </div>
    )
  }

  if (label && !content) {
    return (
      <div className={defaultClassNames}>
        <Ui_Dropdown.Label tabIndex={tabIndex || 0} className="btn m-1">
          {label}
        </Ui_Dropdown.Label>
        {children}
      </div>
    )
  }

  if (label && content) {
    return (
      <div className={defaultClassNames} ref={ref} style={style}>
        <Ui_Dropdown.Label
          tabIndex={0}
          className={`btn m-1 
          btn${' '}
        ${primary ? 'btn-primary' : ''}
        ${secondary ? 'btn-secondary' : ''}
        ${warning ? 'btn-warning' : ''}
        ${info ? 'btn-info' : ''}
        ${success ? 'btn-success' : ''}
        ${error ? 'btn-error' : ''}
        ${accent ? 'btn-accent' : ''}
        ${ghost ? 'btn-ghost' : ''}
        ${link ? 'btn-link' : ''}
        ${neutral ? 'btn-neutral' : ''}
        ${outline ? 'btn-outline' : ''}
        ${loading ? 'loading' : ''}
        ${wide ? 'btn-wide' : ''}
        ${square ? 'btn-square' : ''}
        ${circle ? 'btn-circle' : ''}
        ${disabled ? 'btn-disabled' : ''}
        ${medium ? 'btn-md' : ''}
        ${small ? 'btn-sm' : ''}
        ${tiny ? 'btn-xs' : ''}
        ${large ? 'btn-lg' : ''}
        ${noAnimation ? 'no-animation' : ''}
        ${block || full ? 'btn-block' : ''}`.trim()}
        >
          {label}
        </Ui_Dropdown.Label>
        <Ui_Dropdown.Content tabIndex={0} className="w-52">
          {content}
        </Ui_Dropdown.Content>
        {children}
      </div>
    )
  }

  return <></>
})

const Label: FC = forwardRef<HTMLLabelElement, Ui_DropdownProps>(({ children, className, tabIndex }, ref) => {
  return (
    <label tabIndex={tabIndex || 0} className={`${className ? className : ''}`} ref={ref}>
      {children}
    </label>
  )
})

const Content: FC = forwardRef<HTMLDivElement, Ui_DropdownProps>(({ children, className }, ref) => {
  const classNamesArray = className?.split(' ') || ([] as string[])
  const hasBgClass = classNamesArray.find(t => t.includes('bg-'))

  let defaultClasses = `dropdown-content rounded-box shadow-2xl border border-base-content border-opacity-5`
  const defaultBgClass = ' bg-base-100 '

  if (hasBgClass) defaultClasses += hasBgClass
  if (!hasBgClass) defaultClasses += defaultBgClass

  return (
    <div className={`${defaultClasses} ${className ? className : ''}`} ref={ref}>
      {children}
    </div>
  )
})

Ui_Dropdown.Label = Label
Ui_Dropdown.Content = Content

export default Ui_Dropdown
