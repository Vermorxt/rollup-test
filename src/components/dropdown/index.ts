export { default } from './dropdown'
export type { Ui_DropdownProps } from './type'
