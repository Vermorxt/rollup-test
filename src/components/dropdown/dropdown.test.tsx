import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { Ui_DropdownProps } from './type'
import { Ui_Dropdown } from './dropdown'
import React from 'react'
import { checkAccessibility, itSupportsSystemProps } from '../../_tests-utils'

const defaultProps: Ui_DropdownProps = {}

const selector = 'dropdown'

describe('@cms/core/Ui_Dropdown', () => {
  checkAccessibility([
    <Ui_Dropdown {...defaultProps} ref={null}>
      Test
    </Ui_Dropdown>,
  ])
  itSupportsSystemProps({
    component: Ui_Dropdown,
    props: defaultProps,
    refType: HTMLDivElement,
  })

  it('should match snapshot', () => {
    const { container } = render(
      <Ui_Dropdown {...defaultProps} ref={null}>
        Test
      </Ui_Dropdown>
    )
    expect(container).toMatchSnapshot()
  })
  // NOTE: (possible debug): debugComponent(<Ui_Card.Title {...defaultProps} ref={null} />)

  it('renders default class', () => {
    const { container: available } = render(<Ui_Dropdown {...defaultProps} ref={null} />)

    expect(available.querySelectorAll(`.${selector}`)).toHaveLength(1)
  })
})
