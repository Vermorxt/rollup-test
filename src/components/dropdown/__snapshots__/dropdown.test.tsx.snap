// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@cms/core/Ui_Dropdown should match snapshot 1`] = `
<div>
  <div
    class="dropdown   "
  >
    Test
  </div>
</div>
`;
